### PR TITLE
Fix FP with Photos, Circles, Notes, Contacts, Files and Settings

### PR DESCRIPTION
**Pull request by EsadCetiner**
This pull request fixes the following
1. Minor formatting issues regarding CRS guidelines, I used the CRS rules check tool to find these issues.
2. False positives when changing settings for Photos, Contacts, Notes and Shares.
3. Fixed false positive when creating/deleting/modifying Circles.

### CODE REVIEW

**Comment by theseion** [comment]
Thanks @EsadCetiner. There's a similar PR #28. AFAICT yours is a super set of the other one. Can you confirm that?

**Comment by EsadCetiner** [comment]
@theseion I've taken a look at it and everything looks good to me. although I am not sure why``ctl:ruleRemoveByTag=attack-injection-php,\`` for rule ``9508341`` is needed for or why for rule ``9508126`` was changed from `` ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:autosaveContent"`` ``ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS"``. I haven't encountered those false positives specifically.

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -219,46 +219,47 @@ SecRule REQUEST_FILENAME "@contains /apps/recommendations/settings/enabled" \
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
     setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
 
-# Text app autosave sync feature doesn't works 
+# Text app autosave sync feature doesn't works
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:222]:
```suggestion
# Text app autosave sync feature doesn't work
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 2]
```diff
@@ -219,46 +219,47 @@ SecRule REQUEST_FILENAME "@contains /apps/recommendations/settings/enabled" \
...
 SecRule REQUEST_URI "@contains /apps/text/session/sync" \
     "id:9508126,\
     phase:1,\
     pass,\
     t:none,\
     nolog,\
-    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:autosaveContent"
+    ctl:ruleRemoveTargetById=941100;ARGS:json.documentState,\
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:autosaveContent,\
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:230]: Please add a comment to the rule to document which argument names concern which versions of Nextcloud.

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 3]
```diff
@@ -408,6 +409,15 @@ SecRule REQUEST_METHOD "@streq PUT" \
         "t:none,\
         ctl:ruleRemoveById=200002"
 
+# When enabing "Update avatar from social media" setting in Nextcloud Contacts
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:412]:
```suggestion
# When enabling "Update avatar from social media" setting in Nextcloud Contacts
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 4]
```diff
@@ -845,36 +855,86 @@ SecRule REQUEST_FILENAME "@rx /index\.php/apps/notes/api/v[0-9]/notes" \
...
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.title"
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.title,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0'"
 
 # Note name and content can be anything
 # PUT - when making changes/creating notes
 # DELETE - when notes are deleted
 SecRule REQUEST_FILENAME "@contains /apps/notes/notes/" \
-    "id:9508920,\
+    "id:9508911,\
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:866]: Why did you change the ID?

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 5]
```diff
@@ -845,36 +855,86 @@ SecRule REQUEST_FILENAME "@rx /index\.php/apps/notes/api/v[0-9]/notes" \
...
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
 #
 # [ Custom CSS ]
 #
 
 # Allow admin to add custom CSS code
 SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/provisioning_api/api/v[0-9]/config/apps/theming_customcss/customcss$" \
-   "id:9508930,\
+    "id:9508920,\
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:892]: Why did you change the ID?

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 6]
```diff
@@ -845,36 +855,86 @@ SecRule REQUEST_FILENAME "@rx /index\.php/apps/notes/api/v[0-9]/notes" \
...
+    t:none,\
+    nolog,\
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.value,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0'"
+
+#
+# [ Nextcloud Photos ]
+#
+
+# When changin settings for Nextcloud photos
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:930]:
```suggestion
# When changing settings for Nextcloud photos
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 1 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf]: fixed

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 3 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf]: fixed

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 4 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:866]: I changed it to make room for new rule exclusions, more of a cleanliness thing.

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 2 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:231]: fixed

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 6 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf]: fixed

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 4 above)
Comment (theseion) [reply plugins/nextcloud-rule-exclusions-before.conf:867]: I like that, in general. What I'm not sure of is how stable we need to keep the IDs of individual rules. @dune73, any thoughts?

Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf]: I've reverted the rule ID changes for now, this could be done in a separate PR

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 7]
```diff
@@ -699,7 +710,77 @@ SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/apps/provisioning_api/api/v[0-9]
...
+    "id:9508620,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:mail_smtppassword,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0'"
+
+# When removing whitelisted IPs under Administration --> Security --> Brute Force IP whitelsit
+SecRule REQUEST_FILENAME "@rx /apps/bruteforcesettings/ipwhitelist/[0-9]*$" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:765]: Is there `DELETE` case when the last path segment is not a number?

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 8]
```diff
@@ -858,23 +939,117 @@ SecRule REQUEST_FILENAME "@contains /apps/notes/notes/" \
...
+    "id:9508940,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT DELETE'"
+
+# When setting a unique password for all shares to a circle
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/.*/setting$" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:988]:
```suggestion
SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/.+/setting$" \
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 9]
```diff
@@ -858,23 +939,117 @@ SecRule REQUEST_FILENAME "@contains /apps/notes/notes/" \
...
+    "id:9508962,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
+# Rejecting a file ownership transfer
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/files/api/v[0-9]/transferownership/[0-9]*$" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1046]: Is there `DELETE` case when the last path segment is not a number?

**Review by theseion — CHANGES_REQUESTED** [review]

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 7 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:765]: No, the number is needed to distinguish between multiple entries for whitelisted IP addresses.

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 9 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:1046]: No, it's needed to distinguish between multiple file ownership transfer requests.

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 7 above)
Comment (theseion) [reply plugins/nextcloud-rule-exclusions-before.conf:765]: So rather `SecRule REQUEST_FILENAME "@rx /apps/bruteforcesettings/ipwhitelist/[0-9]+$" \` (`+` instead of `*`)?

**Review by theseion — CHANGES_REQUESTED** [review]

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 10]
```diff
@@ -858,23 +939,117 @@ SecRule REQUEST_FILENAME "@contains /apps/notes/notes/" \
...
+    "id:9508940,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT DELETE'"
+
+# When setting a unique password for all shares to a circle
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/.+/setting$" \
```
[the remark this answers is not in the document]
Comment (theseion) [reply plugins/nextcloud-rule-exclusions-before.conf:988]: Sorry, no. This: `SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/[^/]+/setting$"`. `.+` could be `some/random/segments/123`, for example.

[the remark this answers is not in the document]
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:988]: it works, I'll commit the change.

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Thans a lot @EsadCetiner!
